src/snowflake/cli/plugins/connection/commands.py

```diff
@@ -213,13 +215,64 @@ def add(
 @app.command()
 @global_options
 @with_output
-def test(connection: str = ConnectionOption, **options) -> CommandResult:
+def test(
+    connection: str = ConnectionOption,
+    enable_diag: bool = typer.Option(
+        False,
+        "--enable-diag",
+        help="Run python connector diagnostic test.",
+        show_default=False,
+    ),
+    diag_log_path: str = typer.Option(
```

Comment:
```suggestion
    diag_log_path: Path = typer.Option(
```
We can also use `typer.Option` arguments `file_ok` and `dir_ok` to do early evaluation if the path is valid one

Reply:
Done